tests/test_zarr_write.py

```diff
@@ -0,0 +1,151 @@
+    h5f.create_dataset('dset_float64', data=np.array([1, 2, 3], dtype=np.float64))
+    h5f.create_dataset('dset_bool', data=np.array([True, False, True], dtype=np.bool_))
+
+    group1 = h5f.create_group('group1')
+    group1.attrs['attr_str'] = 'hello'
+    group1.attrs['attr_int'] = 42
+    group1.create_dataset('dset_with_nan', data=np.array([1, np.nan, 3], dtype=np.float64))
+    group1.create_dataset('dset_with_inf', data=np.array([np.inf, 6, -np.inf], dtype=np.float64))
+
+    compound_dtype = np.dtype([('x', np.int32), ('y', np.float64)])
+    group1.create_dataset('dset_compound', data=np.array([(1, 2.2), (3, 4.4)], dtype=compound_dtype))
+
+    group_to_delete = h5f.create_group('group_to_delete')
+    group_to_delete.attrs['attr_str'] = 'hello'
+    group_to_delete.attrs['attr_int'] = 42
+    group_to_delete.create_dataset('dset_to_delete', data=np.array([1, 2, 3], dtype=np.int8))
+    del h5f['group_to_delete']
+
+    another_group_to_delete = group1.create_group('another_group_to_delete')
+    another_group_to_delete.attrs['attr_str'] = 'hello'
+    another_group_to_delete.attrs['attr_int'] = 42
+    another_group_to_delete.create_dataset('dset_to_delete', data=np.array([1, 2, 3], dtype=np.int8))
+    del group1['another_group_to_delete']
+
+    yet_another_group_to_delete = group1.create_group('yet_another_group_to_delete')
+    yet_another_group_to_delete.attrs['attr_str'] = 'hello'
+    yet_another_group_to_delete.attrs['attr_int'] = 42
+    yet_another_group_to_delete.create_dataset('dset_to_delete', data=np.array([1, 2, 3], dtype=np.int8))
+    del h5f['group1/yet_another_group_to_delete']
+
+
+def compare_example_h5_data(h5f: h5py.File, tmpdir: str):
+    with h5py.File(f'{tmpdir}/for_comparison.h5', 'w') as h5f2:
+        write_example_h5_data(h5f2)
+    with h5py.File(f'{tmpdir}/for_comparison.h5', 'r') as h5f2:
+        _assert_groups_equal(h5f, h5f2)
+
+
+def _assert_groups_equal(h5f: h5py.Group, h5f2: h5py.Group):
+    print(f'Comparing groups: {h5f.name}')
+    _assert_attrs_equal(h5f, h5f2)
+    for k in h5f.keys():
+        X1 = h5f[k]
+        X2 = h5f2[k]
+        if isinstance(X1, h5py.Group):
+            assert isinstance(X2, h5py.Group)
+            _assert_groups_equal(X1, X2)
+        elif isinstance(X1, h5py.Dataset):
+            assert isinstance(X2, h5py.Dataset)
+            _assert_datasets_equal(X1, X2)
+        else:
+            raise Exception(f'Unexpected type: {type(X1)}')
+
+    for k in h5f2.keys():
+        if k not in h5f:
+            raise Exception(f'Key {k} not found in h5f')
+
+
+def _assert_datasets_equal(h5d1: h5py.Dataset, h5d2: h5py.Dataset):
+    print(f'Comparing datasets: {h5d1.name}')
+    assert h5d1.shape == h5d2.shape, f'h5d1.shape: {h5d1.shape}, h5d2.shape: {h5d2.shape}'
+    assert h5d1.dtype == h5d2.dtype, f'h5d1.dtype: {h5d1.dtype}, h5d2.dtype: {h5d2.dtype}'
+    assert h5d1.dtype == h5d2.dtype, f'h5d1.dtype: {h5d1.dtype}, h5d2.dtype: {h5d2.dtype}'
```

Comment:
```suggestion
    assert h5d1.name == h5d2.name, f'h5d1.name: {h5d1.name}, h5d2.name: {h5d2.name}'
```

It might also be useful to compare the other important dataset properties like compression, chunks, fill value, fletcher32, etc. but that can be done in a separate PR.